src/boot/boot.c

```diff
@@ -1787,7 +1794,11 @@ static bool config_sysfail_occured(Config *config) {
 
         assert(config);
 
-        sysfail_type = sysfail_check();
+        SysFailConfig sysfail_config = {
+                .check_firmware_update = config->sysfail_firmware_update
+        };
```

Comment:
why make this configurable? seems unnecessary to me? i mean the SysFail entry is not going to be set by default, hence we aren't going to do anything in the default case anyway, hence seems safe to just enable this